sentry-spring-boot-starter/src/main/java/io/sentry/spring/autoconfigure/SentryAutoConfiguration.java

```diff
@@ -41,4 +54,79 @@ public ServletContextInitializer sentryServletContextInitializer() {
         return new SentryServletContextInitializer();
     }
 
+    /**
+     * Initializes a {@link SentryClient}.
+     *
+     * @return a new instance of {@link SentryClient}.
+     */
+    @Bean
+    @ConditionalOnMissingBean(SentryClient.class)
+    @ConditionalOnProperty(name = "sentry.init-default-client", havingValue = "true", matchIfMissing = true)
+    public SentryClient sentryClient(SentryProperties properties,
+                                     @Autowired(required = false) List<EventBuilderHelper> eventBuilderHelpers,
+                                     @Autowired(required = false) List<EventSendCallback> eventSendCallbacks,
+                                     @Autowired(required = false) List<ShouldSendEventCallback> shouldSendEventCallbacks) {
+        if (properties.getOptions() != null && !properties.getOptions().isEmpty()) {
```

Comment:
Have you considered manually building a `Lookup` instance using a custom `ConfigurationProvider` and using it like:

```java
class MapConfigurationProvider implements ConfigurationProvider {
  @Override
  public String getProperty(String key) {
    return properties.get(key);
  }
}

class EmptyConfigurationProvider implements ConfigurationProvider {
  @Override
  public String getProperty(String key) {
    return null;
  }
}

Lookup myCustomLookupInstance = new Lookup(new MapConfigurationProvider(), new EmptyConfigurationProvider());

SentryClient client = Sentry.init(SentryOptions.from(myCustomLookupInstance));
``` 

That way we wouldn't have to set system properties (which is guarded by a security manager - even if that is not probably commonly used in spring boot apps).